web/src/main/kotlin/com/example/ajaxproject/service/EmailSenderServiceImpl.kt

```diff
@@ -9,29 +9,37 @@ import org.slf4j.LoggerFactory
 import org.springframework.mail.javamail.JavaMailSender
 import org.springframework.mail.javamail.MimeMessageHelper
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
 
 @Service
 internal class EmailSenderServiceImpl(private val javaMailSender: JavaMailSender) : EmailSenderService {
 
-    override fun send(emailDTO: EmailDTO): SendEmailResponse {
-        return runCatching {
+    val emailRegex = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\$"
+
+    override fun send(emailDTO: EmailDTO): Mono<SendEmailResponse> {
+        return Mono.defer {
             javaMailSender.send(generateMailMessage(emailDTO))
             logger.debug("Email sent successfully")
-            SendEmailResponse(status = 200)
-        }.getOrElse { exception ->
-            logger.warn("An error has occurred sending email", exception)
-            SendEmailResponse(status = 500, cause = exception.message)
+            Mono.just(SendEmailResponse(status = 200))
+        }.onErrorResume { error ->
+            logger.error("An unexpected error has occurred sending email", error)
+            Mono.just(SendEmailResponse(status = 500, cause = error.message))
         }
     }
-    private fun generateMailMessage(emailDTO: EmailDTO): MimeMessage {
 
+    private fun generateMailMessage(emailDTO: EmailDTO): MimeMessage {
         return MimeMessageHelper(javaMailSender.createMimeMessage()).apply {
             setFrom(emailDTO.from)
             setTo(emailDTO.to)
-            emailDTO.subject?.let { setSubject(it) }
+            emailDTO.subject.let { setSubject(it) }
             setText(emailDTO.body, true)
         }.mimeMessage
     }
+
+    override fun isValidEmail(email: String): Boolean {
+        return email.matches(emailRegex.toRegex())
```

Comment:
It's better to compile regex once and reuse the compiled result.

Compiling of regex is heavy operation.

So, in companion object:
```
val EMAIL_REGEX: <TYPE> = "your patter here".toRegex()
```